src/UI/Components/Diagram/helpers.ts

```diff
@@ -529,3 +531,71 @@ export const toggleLooseElement = (
     }),
   );
 };
+
+/**
+ * Gets the coordinates of all cells in the graph.
+ *
+ * @param {dia.Graph} graph - The graph from which to get the cells.
+ * @returns {SavedCoordinates[]} An array of objects, each containing the id, name, attributes, and coordinates of a cell.
+ */
+export const getCellsCoordinates = (graph: dia.Graph): SavedCoordinates[] => {
+  const cells = graph.getCells();
+  return cells
+    .filter((cell) => cell.attributes.type === "app.ServiceEntityBlock")
+    .map((cell) => ({
+      id: cell.id,
+      name: cell.attributes.entityName,
+      attributes: cell.attributes.instanceAttributes,
+      coordinates: cell.attributes.position,
+    }));
+};
+
+/**
+ * Applies coordinates to cells in the graph.
+ *
+ * @param {dia.Graph} graph - The graph to which to apply the coordinates.
+ * @param {SavedCoordinates[]} coordinates - The coordinates to apply to the cells.
+ */
+export const applyCoordinatesToCells = (
+  graph: dia.Graph,
+  coordinates: SavedCoordinates[],
+) => {
+  const cells = graph.getCells();
+  coordinates.forEach((element) => {
+    const correspondingCell = cells.find(
+      (cell) =>
+        element.id === cell.id ||
+        (element.name === cell.attributes.entityName &&
+          isEqual(element.attributes, cell.attributes.instanceAttributes)),
+    );
+    if (correspondingCell) {
+      correspondingCell.set("position", {
+        x: element.coordinates.x,
+        y: element.coordinates.y,
+      });
+    }
+  });
+};
+
+/**
+ * Moves a cell away from any cells it is colliding with.
+ *
+ * @param {dia.Graph} graph - The graph containing the cell.
+ * @param {dia.Cell} cell - The cell to move.
+ */
+export const moveCellFromColliding = (graph: dia.Graph, cell: dia.Cell) => {
+  let isColliding = false;
+  do {
+    const elementsUnder = graph
```

Comment:
what's an elementsUnder?

Reply:
changed to `elementsUnderTheCell`

Reply:
But what is underTheCell then ? Is it like, connected to it? A decendant? Is it visually underneath ?

Reply:
The docstrings explains that this function moves cell away from any cell it is colliding with, so yes, visually

Reply:
Then overlapping elements is probably the term you need :D or simply collidingElements